### DISCUSSION TITLE
No way to disable the focus ring globally?

### DISCUSSION THREAD

**Discussion by lcswillems** [chakra-ui/chakra-ui#9251] [Q&A] [answered]
In v2 it was possible to disable the focus ring globally. But it doesn't seem feasible in v3?

https://www.chakra-ui.com/docs/get-started/migration

**Comment by ELHart05** [comment]
No, you can normally disable it, I'm using `"@chakra-ui/react": "^3.2.1"` and `"@emotion/react": "^11.13.5"` and it's working just fine with this
```jsx
    <Input
      width='50%'
      ml={"200px"}
      mt={"200px"}
      focusVisibleRing={'none'}
      borderColor={'gray.300'}
    />
```

**Reply by zanlib0** [reply]
How do you do this globally?

**Reply by lcswillems** [reply]
Yes, how to do it globally? That's the thing I'm looking for.

**Comment by ELHart05** [comment]
This is the solution I reached,

- Create a file to extend the theme,

```jsx
import {
  createSystem,
  defaultConfig,
  defineConfig,
  defineRecipe
} from "@chakra-ui/react"

const inputReceipt = defineRecipe({
  base: {
    border: '1px solid black !important', //override the border property by this
    outline: 'none !important',
    focusRing: 'none !important',
  },
})

const customConfig = defineConfig({
  theme: {
    recipes: {
      input: inputReceipt,
    },
  },
})


const system = createSystem(customConfig, defaultConfig);

export default system;
```

- Call the system theme as the "value" for the value prop passed to ChakraProvider like this `<ChakraProvider value={system}>`

<hr />

Another way to use it globally is to create a custom input that wraps the original chakra input with the props that removes the focus ring like this>

```jsx
const MyInput = ({ focusVisibleRing, borderColor...rest }) => {
   return (
        <Input
          focusVisibleRing={focusVisibleRing ?? 'none'}
          borderColor={borderColor ?? 'gray.300'}
          {...rest}
        />
   )
}
```

**Reply by ELHart05** [reply]
@lcswillems I hope it helped

**Comment by lucasishuman** [comment]
Focus ring docs ~
https://www.chakra-ui.com/docs/styling/focus-ring

**Reply by lcswillems** [reply]
@lucasishuman Does it explain how to disable it?

**Reply by jojoalmost** [reply]
@lcswillems 

this my solution, hope can help you

```

import { createSystem, defaultConfig, defineConfig } from '@chakra-ui/react';

const config = defineConfig({
    globalCss: {
    html: {
      focusVisibleRing: 'none',
      focusRingWidth: 0,
    },
  },
});

export const system = createSystem(defaultConfig, config);


```

**Comment by segunadebayo** [comment, marked as answer]
Set `focusVisibleRing: "none"` on the recipe, or `focusVisibleRing="none"` on the instance.

https://chakra-ui.com/docs/styling/focus-ring
